client/src/components/App.tsx

```diff
@@ -0,0 +1,212 @@
+import React, { useState, useEffect } from "react";
+import { createMuiTheme, makeStyles } from "@material-ui/core/styles";
+import { Button, Typography } from "@material-ui/core";
+import { createSession, SuccessfulCreationResponse } from "api";
+import ChatThread from "components/ChatThread";
+import ChatForm from "components/ChatForm";
+import { TargetIndicator } from "components/TargetIndicator";
+import SummaryPopup from "components/SummaryPopup";
+import ErrorPopup from "components/ErrorPopup";
+import withLocation from "wrap-with-location";
+import { errorForStatus } from "components/ErrorConfig";
+
+const theme = createMuiTheme({
+  palette: {
+    primary: {
+      main: "#1b6a9c",
+    },
+  },
+});
+
+const useStyles = makeStyles((theme) => ({
+  foreground: {
+    backgroundColor: theme.palette.primary.main,
+    width: "100%",
+    height: "calc(100% - 75px)",
+    minHeight: 750,
+    position: "absolute",
+    left: "50%",
+    transform: "translate(-50%, 0%)",
+  },
+  image: {
+    width: "100%",
+    height: "30%",
+    aspectRatio: "2/1",
+    objectFit: "cover",
+    marginBottom: -5,
+  },
+  chatWindow: {
+    backgroundColor: theme.palette.background.default,
+    width: "100%",
+    height: "67%",
+    position: "absolute",
+    left: "50%",
+    transform: "translate(-50%, 0%)",
+    marginBottom: 15,
+  },
+  buildInfo: {
+    position: "fixed",
+    bottom: 4,
+    marginLeft: 8,
+    color: "white",
+    fontWeight: "bold",
+    fontSize: "70%",
+  },
+}));
+
+const App = (props: { search: { lesson: string } }) => {
+  const styles = useStyles();
+  const { lesson } = props.search;
+  const [summaryOpen, setSummaryOpen] = React.useState(false);
+  const [summaryMessage, setSummaryMessage] = React.useState(
+    "Let's see how you're doing so far!"
+  );
+  const [targets, setTargets] = React.useState<
+    {
+      achieved: boolean;
+      score: number;
+      text: string;
+      status: string;
+    }[]
+  >([]);
+  const [session, setSession] = React.useState({
+    sessionId: "",
+    sessionHistory: "",
+    previousUserResponse: "",
+    previousSystemResponse: [""],
+    dialogState: {
+      expectationsCompleted: false,
+      expectationData: [{ ideal: "", score: 0, satisfied: false, status: "" }],
+      hints: false,
+    },
+    hash: "",
+  });
+
+  const [messages, setMessages] = useState([
+    {
+      senderId: "system",
+      type: "opening",
+      text: "Welcome to OpenTutor!",
+    },
+  ]);
+  const [errorProps, setErrorProps] = useState({
```

Comment:
better to have a non-anonymous type here so it can be shared w the consumer[s] of this state